commit 156cf14561ba750503e15b79f128b240e77b4991
Author: Stéphane Graber <stgraber@ubuntu.com>
Date:   Sat Jul 1 00:38:51 2017 -0400

    network: Don't fail on non-process PIDs
    
    If the PID we have point to a kernel thread, /proc/PID/exe will not
    exist and resolving the symlink will therefore fail.
    
    In such cases, just go ahead with an empty path which will result in the
    right behavior.
    
    Launchpad: https://bugs.launchpad.net/ubuntu/+source/lxd/+bug/1698712
    Signed-off-by: Stéphane Graber <stgraber@ubuntu.com>

diff --git a/lxd/networks_utils.go b/lxd/networks_utils.go
index 3bac33d..098487c 100644
--- a/lxd/networks_utils.go
+++ b/lxd/networks_utils.go
@@ -681,7 +681,7 @@ func networkKillDnsmasq(name string, reload bool) error {
 	// Check if it's dnsmasq
 	cmdPath, err := os.Readlink(fmt.Sprintf("/proc/%s/exe", pid))
 	if err != nil {
-		return err
+		cmdPath = ""
 	}
 
 	// Deal with deleted paths
